backends/arm/test/common.py

```diff
@@ -74,19 +74,15 @@ def get_tosa_compile_spec_unbuilt(
     the compile spec before calling .build() to finalize it.
     """
     if not custom_path:
-        intermediate_path = maybe_get_tosa_collate_path() or tempfile.mkdtemp(
-            prefix="arm_tosa_"
-        )
-    else:
-        intermediate_path = custom_path
+        custom_path = maybe_get_tosa_collate_path()
```

Comment:
Since we don't have to save files unless we want to with this change, we don't need to create a temporary dir when custom_path is not -> I viewed this as a related change.